src/controllers/reviewController.ts

```diff
@@ -0,0 +1,74 @@
+import { Request, Response } from "express";
+import Review from "../models/Review";
+
+export const createReview = async (req: Request, res: Response) => {
+    const { content, postId, parentReview } = req.body;
+    const userId = req.user.userId;
+
+    if (!userId) {
+        return res.status(401).send({ error: 'Unauthorized' });
+    }
+
+    try {
+        const review = new Review({
+            userId,
+            postId,
+            content,
+            parentReview
+        });
+
+        if (parentReview) {
+            const parent = await Review.findById(parentReview);
+            if (parent) {
+                parent.childrenReview.push(review._id);
+                await parent.save();
+            }
+        }
+
+        await review.save();
+        res.status(201).send(review);
+    }
+    catch (error) {
+        res.status(400).send(error);
+    }
+};
+
+export const updateReview = async (req: Request, res: Response) => {
+    const { id } = req.params;
+    const { content } = req.body;
+    const userId = req.user.userId;
+
+    if (!userId) {
```

Comment:
Have you checked if the current user is not the same as the author of this review